src/client/bidding/txBuilder.ts

```diff
@@ -0,0 +1,95 @@
+import { TransactionBlock } from "@mysten/sui.js";
+import { TransactionBlockArgument, TransactionResult, txObj as txCommon } from "../../transaction";
+import { GlobalParams } from "../types";
+import { DEFAULT_BIDDING_MODULE, DEFAULT_PACKAGE_ID } from "../consts";
+import { CloseBidParams, CreateBidInput, CreateBidWithCommissionInput, SellNft, SellNftFromKiosk } from "./types";
+
+const txObj = (
+  fun: string,
+  p: GlobalParams,
+  args: (
+    tx: TransactionBlock
+  ) => (TransactionBlockArgument | TransactionResult)[],
+  tArgs: string[]
+): [TransactionBlock, TransactionResult] => {
+  return txCommon(
+    {
+      packageObjectId: p.packageObjectId ?? DEFAULT_PACKAGE_ID,
+      moduleName: p.moduleName ?? DEFAULT_BIDDING_MODULE,
+      fun,
+      transaction: p.transaction,
+    },
+    args,
+    tArgs
+  );
+}
+
+
+export const createBidTx = (p: CreateBidInput) => {
+    return txObj(
+        "create_bid",
+        p,
+        (tx) => [
+            tx.object(p.nft), // The parameters are reversed because of a smart contract bug, change it when the fixed version will deployed
+            typeof p.buyersKiosk === "string" ? tx.object(p.buyersKiosk) : p.buyersKiosk,
```

Comment:
I see this ternary many times, how about wrap it with function, like `wrapToObject` or something like that??